### PR TITLE
Fix Linux chain.ChainStatus when multiple layers have errors

### PR DESCRIPTION
**Pull request by bartonjs**
The recent chain rewrite caused only the last layer with errors to be reported.

### CODE REVIEW

File: src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs [hunk 1]
```diff
@@ -570,7 +570,7 @@ private static ArraySegment<char> Base64UrlEncode(ReadOnlySpan<byte> input)
                     if (elementErrors.HasValue && elementErrors.Value.HasErrors)
                     {
                         List<X509ChainStatus> statusBuilder = new List<X509ChainStatus>();
-                        overallStatus = new List<X509ChainStatus>();
+                        overallStatus = overallStatus ?? new List<X509ChainStatus>();
```
Comment (stephentoub) [inline src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs]: Nit: I think you can now write this as:
```C#
overallStatus ??= new List<X509ChainStatus>();
```
if you want.

**Comment by bartonjs** [comment]
Looks like I need to move some things around for the netstandard test build, since it doesn't have RandomNumberGenerator.Fill or the CertificateRequest class.

**Comment by bartonjs** [comment]
The Mission Control link says that all the tests passed, but infrastructure got a bunch of network errors.  Let's try one more time.

/azp run corefx-ci

**Pull request merged**